File: programmers/Lv2/12909. 올바른 괄호/올바른 괄호.swift
import Foundation

func solution(_ s:String) -> Bool
{
    var leftCount = 0
    for c in s {
        if c == "(" { leftCount += 1 }
        else { 
            if leftCount > 0 { leftCount -= 1 }
            else { return false }
        }
    }
    return leftCount > 0 ? false : true
}

File: programmers/Lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    left_count = 0
    right_count = 0
    for c in s:
        if c == "(":
            left_count += 1
        elif c == ")":
            right_count += 1
        if right_count > left_count: return False

    return left_count == right_count